.github/workflows/generate-changelog.yaml

```diff
@@ -0,0 +1,127 @@
+    runs-on: ubuntu-22.04
+    outputs:
+      content: ${{ steps.read-config.outputs.content }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          # required for getting all tags
+          fetch-depth: 0
+          # required for bypassing status checks when pushing
+          token: "${{ secrets.github-token }}"
+
+      - name: Read changelog config
+        id: read-config
+        # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+        run: |
+          content=$(cat "${{ inputs.config }}")
+          { echo "content<<EOF"; echo "$content"; echo "EOF"; } >> "$GITHUB_OUTPUT"
+        shell: bash
+
+  generate-changelog:
+    name: Generate Changelog
+    runs-on: ubuntu-22.04
+    needs: [read-config-file]
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          # required for getting all tags
+          fetch-depth: 0
+          # required for bypassing status checks when pushing
+          token: "${{ secrets.github-token }}"
+
+      - name: Generate Changelog
+        id: generate-release-changelog
+        uses: heinrichreimer/github-changelog-generator-action@v2.2
+        with:
+          token: "${{ secrets.github-token }}"
+          sinceTag: "${{ inputs.old-tag }}"
+          output: "changes.md"
+          stripGeneratorNotice: "true"
+          issues: "true"
+          unreleased: "true"
+          pullRequests: "false"
+          compareLink: "true"
+          issuesWoLabels: "false"
+          futureRelease: "${{ inputs.new-tag }}"
+          issuesLabel: "**Miscellaneous:**"
+          bugLabels: "type/bug"
+          enhancementLabels: "type/enhancement"
+          httpCache: "true"
+          verbose: "true"
+          addSections: |
+            ${{ needs.read-config-file.outputs.content }}
+
+      - name: Update github Changelog
+        run: |
+          # Remove header from changes
+          sed -i '1d' changes.md
+          # Prepend changes.md after first header
+          sed -i '/# Changelog/ r changes.md' "${{ inputs.gh-changelog }}"
+        shell: bash
+
+      - name: Update doc Changelog
+        run: |
+          if [ -n "${{ inputs.doc-changelog }}" ]; then 
+            sed -i '1d' changes.md
+            sed -i '/# Changelog/ r changes.md' "${{ inputs.doc-changelog }}"
+          fi
+        shell: bash
+
+      - name: Create release commit
+        run: |
+          # Set email and username for github-actions bot
+          git config --local user.email "${{ secrets.github-email }}"
+          git config --local user.name "${{ secrets.github-token }}"
+          # Add changes
+          git add CHANGELOG.md
+          git add docs/
+          git add deployment/
+          git commit -m "Release v${{ inputs.new-tag }}"
```

Comment:
I'd argue that this action should not create a commit at all (or at least make it optional). A lot of times you want to include other changes in the release commit.

Reply:
I was thinking about that as well, but so far we only have use cases where we commit and push it right away. How would you output it otherwise? Would you output it as a file not being committed or would you use an job outputs?

Reply:
Please remove the git commit and push step. It is fine if you just leave the Changelog artifact(s) in the local file system

Reply:
am resolving this conversation since this is implemented/fixed in the new version